common/src/models/error.rs

```diff
@@ -41,8 +41,12 @@ pub enum ErrorKind {
     UserAlreadyExists,
     ProjectNotFound,
     InvalidProjectName,
-    ProjectAlreadyExists,
-    ProjectAlreadyRunning,
+    ProjectAlreadyExists {
+        // A message describing a running state of the project.
+        // Used if the project already exists but is owned
+        // by the caller, which means they can modify the project.
+        owner_state_msg: Option<String>,
```

Comment:
Hmm, should we change this to `state_msg` or just `message`? And we should be able to make this a doc comment: `///`.

Reply:
Yes, sounds like a good choice. I felt like this name was suboptimal anyways. I tried to make it clear that this is a message which is directly aimed at the project owner but not for any other case. Alternatively the entire `Option` could be replaced by an additional variant in the `enum` too.

Reply:
Ah, maybe we could just have `ProjectAlreadyExists` and have a message field on that? :thinking: